### PR TITLE
Account creation frontend up to sending email

### PR DESCRIPTION
**Pull request by macieksmuga**
Implement frontend for first part of the create user flow, surfacing a dialog on sending the email and a "resend" link.

https://sig-gis.atlassian.net/browse/PLAM-24

Shows errors from server:

image

Confirmation dialog:

image

### CODE REVIEW

**Review by janicejl — APPROVED** [review]

File: src/interface/src/app/services/auth.service.ts [hunk 1]
```diff
@@ -70,12 +70,16 @@ export class AuthService {
         email,
         first_name: firstName,
         last_name: lastName,
-      })
-      .pipe(
-        concatMap((_) => {
```
Comment (sterlingwellscaffeine) [inline src/interface/src/app/services/auth.service.ts:75]: Just double checking that this code was meant to be removed?

File: src/interface/src/app/signup/signup.component.spec.ts [hunk 2]
```diff
@@ -94,15 +94,4 @@ describe('SignupComponent', () => {
       );
     });
   });
-
-  describe('login', () => {
```
Comment (sterlingwellscaffeine) [inline src/interface/src/app/signup/signup.component.spec.ts:98]: Why are we removing the login test?

File: src/interface/src/app/signup/validation-email-dialog/validation-email-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  MatDialog,
+  MatDialogRef,
+} from '@angular/material/dialog';
+import { of } from 'rxjs';
+import { MaterialModule } from 'src/app/material/material.module';
+import { AuthService } from 'src/app/services';
+
+import { ValidationEmailDialog } from './validation-email-dialog.component';
+
+fdescribe('ValidationEmailDialogComponent', () => {
```
Comment (sterlingwellscaffeine) [inline src/interface/src/app/signup/validation-email-dialog/validation-email-dialog.component.spec.ts:16]: Please remove the fdescribe for checked in code

**Review by sterlingwellscaffeine — APPROVED** [review]

File: src/interface/src/app/services/auth.service.ts (hunk 1 above)
Comment (macieksmuga) [reply src/interface/src/app/services/auth.service.ts:75]: That's correct. We're no longer logging in after sending the confirmation email.

File: src/interface/src/app/signup/signup.component.spec.ts (hunk 2 above)
Comment (macieksmuga) [reply src/interface/src/app/signup/signup.component.spec.ts:98]: The signup component no longer performs a login. Its only user-visible action is to send a validation email (although it does create the user account in the background – but that's all done on the backend, and I can't test it from here)

File: src/interface/src/app/signup/validation-email-dialog/validation-email-dialog.component.spec.ts (hunk 3 above)
Comment (macieksmuga) [reply src/interface/src/app/signup/validation-email-dialog/validation-email-dialog.component.spec.ts]: Done. Thank you for catching that!

**Comment by macieksmuga** [comment]
There's a test failure involving a dialog container, TBD

**Pull request merged**